Data/ByteString/Base64/Internal.hs

```diff
@@ -155,18 +155,16 @@ decodeWithTable padding !decodeFP bs
       | r == 3 -> validateLastPad bs noPad $ withBS (B.append bs (B.replicate 1 0x3d)) go
       | otherwise -> Left "Base64-encoded bytestring has invalid size"
   where
-    (!q, !r) = (B.length bs) `divMod` 4
+    !r = B.length bs `rem` 4
 
     noPad = "Base64-encoded bytestring required to be unpadded"
     invalidPad = "Base64-encoded bytestring has invalid padding"
 
-    !dlen = q * 3
-
     go !sptr !slen = do
-      dfp <- mallocByteString dlen
-      withForeignPtr decodeFP $! \ !decptr ->
-        withForeignPtr dfp $! \dptr ->
-          decodeLoop decptr sptr dptr (sptr `plusPtr` slen) dfp
+      dfp <- mallocByteString (slen * 3)
```

Comment:
At this point, the correct number of pads should have been filled out in the string, and `slen * 3` is then correct, since `slen`'s length is now `0` modulo 4.